File: app/src/main/java/com/bitflaker/lucidsourcekit/database/notifications/daos/NotificationCategoryDao.java
package com.bitflaker.lucidsourcekit.database.notifications.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.bitflaker.lucidsourcekit.database.notifications.entities.NotificationCategory;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface NotificationCategoryDao {
    @Query("SELECT * FROM NotificationCategory ORDER BY id")
    Single<List<NotificationCategory>> getAll();

    @Query("SELECT * FROM NotificationCategory WHERE id = :notificationCategoryId")
    Single<NotificationCategory> getById(String notificationCategoryId);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    Completable insertAll(NotificationCategory... NotificationCategories);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    Completable insert(NotificationCategory NotificationCategory);

    @Delete
    Completable delete(NotificationCategory NotificationCategory);

    @Delete
    void deleteAll(List<NotificationCategory> NotificationCategories);

    @Update
    Completable update(NotificationCategory NotificationCategory);
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/database/notifications/daos/NotificationCategoryDao.kt
package com.bitflaker.lucidsourcekit.database.notifications.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import com.bitflaker.lucidsourcekit.database.notifications.entities.NotificationCategory

@Dao
interface NotificationCategoryDao {
    @Query("SELECT * FROM NotificationCategory ORDER BY id")
    suspend fun getAll(): MutableList<NotificationCategory>

    @Query("SELECT * FROM NotificationCategory WHERE id = :notificationCategoryId")
    suspend fun getById(notificationCategoryId: String): NotificationCategory

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertAll(notificationCategories: Array<NotificationCategory>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(notificationCategory: NotificationCategory)

    @Delete
    suspend fun delete(notificationCategory: NotificationCategory)

    @Delete
    suspend fun deleteAll(notificationCategories: List<NotificationCategory>)

    @Update
    suspend fun update(notificationCategory: NotificationCategory)
}
